File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/base/BaseActivityCallback.java
package org.mifos.mobilewallet.mifospay.base;

import android.support.v4.widget.SwipeRefreshLayout;

public interface BaseActivityCallback {

    void showSwipeProgress();

    void hideSwipeProgress();

    void showProgressDialog(String message);

    void hideProgressDialog();

    void cancelProgressDialog();

    void setToolbarTitle(String title);

    void setSwipeRefreshEnabled(boolean enabled);

    void showColoredBackButton(int drawable);

    void showCloseButton();

    void hideBackButton();

    void dismissProgressDialog();

    SwipeRefreshLayout getSwipeRefreshLayout();
}

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/base/BaseActivityCallback.kt
package org.mifos.mobilewallet.mifospay.base

import androidx.swiperefreshlayout.widget.SwipeRefreshLayout

interface BaseActivityCallback {
    fun showSwipeProgress()
    fun hideSwipeProgress()
    fun showProgressDialog(message: String?)
    fun hideProgressDialog()
    fun cancelProgressDialog()
    fun setToolbarTitle(title: String?)
    fun setSwipeRefreshEnabled(enabled: Boolean)
    fun showColoredBackButton(drawable: Int)
    fun showCloseButton()
    fun hideBackButton()
    fun dismissProgressDialog()

    val swipeRefreshLayout: SwipeRefreshLayout?
}